tests/core/gui/test_workspace_sharing.py

```diff
@@ -288,3 +290,48 @@ def _no_workspace_listed():
         "Error",
         translate("TEXT_FILE_SHARING_REVOKED_workspace").format(workspace="Workspace"),
     )
+
+
+@customize_fixtures(logged_gui_as_admin=True)
+@pytest.mark.gui
+@pytest.mark.trio
+async def test_rename_workspace_when_revoked(
+    aqtbot, running_backend, logged_gui, autoclose_dialog, qt_thread_gateway, bob, monkeypatch
+):
+    w_w = await logged_gui.test_switch_to_workspaces_widget()
+
+    core = logged_gui.test_get_tab().core
+    wid = await core.user_fs.workspace_create("Workspace")
+
+    def _workspace_not_shared_listed():
+        assert w_w.layout_workspaces.count() == 1
+        wk_button = w_w.layout_workspaces.itemAt(0).widget()
+        assert isinstance(wk_button, WorkspaceButton)
+        assert wk_button.label_title.text() == "Workspace (private)"
+        assert wk_button.label_title.toolTip() == "Workspace (private)"
+        assert not wk_button.is_shared
+        wk_button.name == "Workspace"
+
+    await aqtbot.wait_until(_workspace_not_shared_listed, timeout=2000)
```

Comment:
timeout need to be overwritten ? (that's kind of fishy to me... ^^)

Reply:
Not really. We have a timer to update the workspace list set to 1s to avoid making to many refreshes. I guess it would be possible to overwrite the timer but I find it easier to just set the timeout to 2s.